src/main/scala/bot/PerChatState.scala

```diff
@@ -1,13 +1,13 @@
 package bot
 
-import bot.domain.AdviseState
+import bot.domain.states.State
 import com.bot4s.telegram.models.{CallbackQuery, Message}
 
-trait PerChatState[F[_]] {
-  def withMessageState(f: F[AdviseState] => F[Unit])(implicit msg: Message, store: AdviseStateStore[F]): F[Unit] =
+trait PerChatState[F[_], S <: State] {
+  def withMessageState(f: F[S] => F[Unit])(implicit msg: Message, store: StateStore[F, S]): F[Unit] =
```

Comment:
Usually these kinds of functions deal with the value directly, instead of passing it wrapped (`S => F[Unit]` and not `F[S] => F[Unit]`)